streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -61,28 +64,42 @@
                  final ChangelogReader changelogReader,
                  final boolean isStandby,
                  final StateDirectory stateDirectory,
-                 final ThreadCache cache) {
+                 final ThreadCache cache,
+                 final StreamsConfig config) {
         this.id = id;
         this.applicationId = applicationId;
         this.partitions = new HashSet<>(partitions);
         this.topology = topology;
         this.consumer = consumer;
         this.cache = cache;
+        eosEnabled = StreamsConfig.EXACTLY_ONCE.equals(config.getString(StreamsConfig.PROCESSING_GUARANTEE_CONFIG));
```

Comment:
We can. But IMHO it's cleaner code like this. We tend to have too many parameters anyway. I would actually remove parameter `applicationId` and extract from `config` too (did not do it in this PR to keep it clean). WDYT?